ldap_jwt_auth/auth/authentication.py

```diff
@@ -0,0 +1,49 @@
+"""
+Module for providing a class for managing authentication.
+"""
+import logging
+
+import ldap
+
+from ldap_jwt_auth.core.config import config
+from ldap_jwt_auth.core.exceptions import InvalidCredentialsError, LDAPServerError
+from ldap_jwt_auth.core.models import UserCredentials
+
+logger = logging.getLogger()
+
+
+class Authentication:
+    """
+    Class for managing authentication against an LDAP server.
+    """
+
+    def authenticate(self, user_credentials: UserCredentials) -> None:
+        """
+        Authenticate a user against an LDAP server based on the provided user credentials.
+        :param user_credentials: The credentials of the user.
+        :raises InvalidCredentialsError: If the user credentials are invalid.
+        :raises LDAPServerError: If there is a problem with the LDAP server.
+        """
+        username = user_credentials.username
+        logger.info("Authenticating a user")
+        logger.debug("Username provided is '%s'", username)
+
+        try:
+            connection = ldap.initialize(config.ldap_server.url)
+            ldap.set_option(ldap.OPT_PROTOCOL_VERSION, 3)
+            ldap.set_option(ldap.OPT_X_TLS_DEMAND, True)
+            ldap.set_option(ldap.OPT_X_TLS_REQUIRE_CERT, ldap.OPT_X_TLS_NEVER)
+            ldap.set_option(ldap.OPT_DEBUG_LEVEL, 0)
+            connection.start_tls_s()
+            connection.simple_bind_s(f"{username}@{config.ldap_server.realm}", user_credentials.password)
```

Comment:
One thing I noticed is that
```
{
	"username": "",
	"password": ""
}
```
also succeeds, this seems to give more info on it
https://stackoverflow.com/questions/10725891/authenticating-to-active-directory-with-python-ldap-always-returns-97
If this was obtained from operations gateway it might have the same issue.

Reply:
lol interesting.. I need to investigate this a bit more.

Reply:
I tried to get this work using what you linked and various other articles but no luck. I am not sure I understand how the search works as It does not return a different response nor does it complain when empty credentials are provided. My knowledge on LDAP is limited so the response is not meaningful to me. I did not want to waste more time on it so instead I decided to modify the endpoint schema class so that it does not allow users to supply empty credentials, see https://github.com/ral-facilities/ldap-jwt-auth/pull/7/commits/a53f571680db1bc8779cf554d8bdb806ca9e6652.